rosbag2_cpp/src/rosbag2_cpp/reindexer.cpp

```diff
@@ -0,0 +1,252 @@
+
+    // Set up reader
+    rosbag2_storage::StorageOptions temp_so = {
+      f_.string(),  // uri
+      storage_options.storage_id,  // storage_id
+      storage_options.max_bagfile_size,  // max_bagfile_size
+      storage_options.max_bagfile_duration,  // max_bagfile_duration
+      storage_options.max_cache_size,  // max_cache_size
+      storage_options.storage_config_uri  // storage_config_uri
+    };
+
+    // We aren't actually interested in reading messages, so use a blank converter option
+    rosbag2_cpp::ConverterOptions blank_converter_options {};
+    ROSBAG2_CPP_LOG_INFO_STREAM("Preparing to open...");
+    bag_reader->open(temp_so, blank_converter_options);
+    ROSBAG2_CPP_LOG_INFO_STREAM("Opened");
+    auto temp_metadata = bag_reader->get_metadata();
+
+    if (temp_metadata.starting_time < metadata_.starting_time) {
+      metadata_.starting_time = temp_metadata.starting_time;
+    }
+    metadata_.duration += temp_metadata.duration;
+    ROSBAG2_CPP_LOG_INFO_STREAM("New duration: " + std::to_string(metadata_.duration.count()));
+    metadata_.message_count += temp_metadata.message_count;
+
+    // Add the topic metadata
+    for (const auto & topic : temp_metadata.topics_with_message_count) {
+      auto found_topic = temp_topic_info.find(topic.topic_metadata.name);
+      if (found_topic == temp_topic_info.end()) {
+        // It's a new topic. Add it.
+        temp_topic_info[topic.topic_metadata.name] = topic;
+      } else {
+        ROSBAG2_CPP_LOG_INFO_STREAM("Found topic!");
+        // Merge in the new information
+        found_topic->second.message_count += topic.message_count;
+        if (topic.topic_metadata.offered_qos_profiles != "") {
+          found_topic->second.topic_metadata.offered_qos_profiles =
+            topic.topic_metadata.offered_qos_profiles;
+        }
+        if (topic.topic_metadata.serialization_format != "") {
+          found_topic->second.topic_metadata.serialization_format =
+            topic.topic_metadata.serialization_format;
+        }
+        if (topic.topic_metadata.type != "") {
+          found_topic->second.topic_metadata.type = topic.topic_metadata.type;
+        }
+      }
+    }
+
+    ROSBAG2_CPP_LOG_INFO("Closing database");
+    bag_reader->reset();
+  }
+
+  // Convert the topic map into topic metadata
+  for (auto & topic : temp_topic_info) {
+    metadata_.topics_with_message_count.emplace_back(topic.second);
+  }
+}
+
+void Reindexer::reindex(const rosbag2_storage::StorageOptions & storage_options)
+{
+  ROSBAG2_CPP_LOG_INFO("Beginning Reindex Operation.");
+
+  // Identify all bag files
+  base_folder_ = storage_options.uri;
+  auto files = get_bag_files(base_folder_);
+  std::cout << "Finished getting database files\n";
+  if (files.empty()) {
+    throw std::runtime_error("No database files found for reindexing. Abort");
+  }
+
+  // Create initial metadata
+  init_metadata(files, storage_options);
+  ROSBAG2_CPP_LOG_INFO_STREAM("Completed init_metadata");
```

Comment:
This doesn't seem INFO-worthy, it's an implementation detail as written. Maybe DEBUG. Or change the text to be something a user without knowledge of the implementation would benefit from.